### DISCUSSION TITLE
TDD operation unreliable

### DISCUSSION THREAD

**Discussion by smunaut** [srsran/srsRAN_Project#19] [General Help] [answered]
Hi all,

I've tried srsRAN_Project theses last few days and I get great and stable results in FDD operation. Running with a B210 on n3  10 MHz wide cell, it's very solid.

However trying the exact same setup running on n41 / n78, I can't get it working. I can sometimes get it to show up on the phone, but rarely and even when it shows up, no way to register.

The config  I used for my `n78` attempts are basically the one from the examples just enabled external clock source and adapted amf params.

Is there something special I should be paying attention to in TDD mode vs FDD mode ?

Cheers,

    Sylvain

PS: Great work btw, my FDD mode experience is much more reliable than when I was using srsRAN gnb.

**Comment by andrepuschmann** [comment, marked as answer]
Thanks @smunaut for your feedback. We've discovered some IOT issues here ourselves with TDD. Working on it.

**Reply by smunaut** [reply]
Ok, thanks for the heads up.

Also, any reason `n78` with 15 kHz SCS isn't supported ?  ( It's not in the `nr_ssb_band_scs_case_table_fr1` table and unfortunately adding it doesn't quite allow it to work, I'm guessing I'm missing something )

**Reply by andrepuschmann** [reply]
n78 and 15khz scs is a difficult combo. in theory it is supported. however, we have not found a COTS UE that actually does it. If you have one, let us know.
if you check the PHY conformance tests for those bands, they also exclude this combo. so in practice it doesn't play a role I believe.

**Reply by smunaut** [reply]
Mmm, why is that a difficult combo ? 

Also, as an update on the origina issue, apparently setting the cell size to 20 MHz helped and now I can connect to the cell much more reliably, both on n41 and n78 ...

**Reply by andrepuschmann** [reply]
> Mmm, why is that a difficult combo?

I really don't know. But could be a DSP complication on the ASIC. For 15kHz data SCS on n78, you would still need to do 30 kHz SSB SCS. If you read the UE caps from the phone (the exchange during initial attach) there is a bitmap of SCS for each band. You see that COTS phones (usually) don't support 15khz SCS on n78. 

> Also, as an update on the origina issue, apparently setting the cell size to 20 MHz helped and now I can connect to the cell much more reliably, both on n41 and n78 ...

Yes, thanks. working on it.

**Reply by mubk00** [reply]
> Also, as an update on the origina issue, apparently setting the cell size to 20 MHz helped and now I can connect to the cell much more reliably, both on n41 and n78 ...

Interesting. Would you mind sharing your setup, such as the COTS UE you are testing with?

**Reply by smunaut** [reply]
@andrepuschmann Why would it be a 30 kHz SCS on SSB ? The case A has support for 3G < freq < 6G, so where would the limitation come from. 

Ok, interesting about the bitmap in the UE caps. Is that captured in the PCAP ?  ATM I'm having a hard time reading those, they don't see to load to anything legible in the latest wireshark 4.03 stable. What should I use to decode/read them ?

@mubk00 I use a "One Plus Nord CE 5G" ( EB2103 ). I also use the "4G switcher" APK to force "NR only". Note that you also have to use sim/network with 001/01 PLMN and also use sim slot 1.

I have a "One Plus 8" ( IN2013 ) and although I can get it to show the network name when on n78, I could never connect to it. I'm actually starting to doubt that this particular phone support 5G SA at all.

**Reply by asheeshtripathi** [reply]
Hi Everyone, I am trying with x310 as gnB, I am able to get gNB up with band 78 in TDD mode with SCS 30 KHz and BW 10 MHz. I modified the DL ARFCN to 639000 to transmit in CBRS band (3580 to 3590 MHz). I first tried with srsUE on n310 but later found out that its not compatible. I then, programmed a SIM card and tried with Telit fn980 5G kit, there I can see the PLMN (99970, that I configured)  but not able to attach to the network, Similarly I tried with, OnePlus 8T, I see 99970 5G in manual network search but can not connect to it. 
In gNB console, I see only a few underflows. I am using open5GS core and it is co-located with gNB in Intel NUC 12. I see communication between gNB and core when I start gNB:
**gNB prints:** 
cci-nuc13@ccinuc13-NUC12DCMi9:~/github/srsRAN_Project/configs$ sudo gnb -c gnb_rf_b200_tdd_n78_10mhz.yml
Available radio types: uhd.

--== srsRAN gNB (commit 77be7d339) ==--

[INFO] [UHD] linux; GNU C++ version 11.2.0; Boost_107400; UHD_4.1.0.5-3
[INFO] [LOGGING] Fastpath logging disabled at runtime.
Making USRP object with args 'type=x300,addr=192.168.40.2,master_clock_rate=184.32e6,send_frame_size=8000,recv_frame_size=8000'
[WARNING] [0/Radio#0] Attempting to set tick rate to 0. Skipping.
Cell pci=1, bw=10 MHz, dl_arfcn=639000 (n78), dl_freq=3585.0 MHz, dl_ssb_arfcn=638976, ul_freq=3585.0 MHz

==== gNodeB started ===
Type <t> to view trace
tLate: 0; Underflow: 1; Overflow: 0;

 **Core prints:** 
03/09 15:34:16.211: [smf] WARNING: PFCP[REQ] has already been associated [127.0.0.7]:8805 (../src/smf/pfcp-sm.c:218)
03/09 15:34:16.211: [upf] WARNING: PFCP[RSP] has already been associated [127.0.0.4]:8805 (../src/upf/pfcp-sm.c:213)
03/09 15:34:20.160: [amf] INFO: gNB-N2 accepted[127.0.0.1]:56318 in ng-path module (../src/amf/ngap-sctp.c:113)
03/09 15:34:20.160: [amf] INFO: gNB-N2 accepted[127.0.0.1] in master_sm module (../src/amf/amf-sm.c:706)
03/09 15:34:20.160: [amf] INFO: [Added] Number of gNBs is now 1 (../src/amf/context.c:1034)
03/09 15:34:20.160: [amf] INFO: gNB-N2[127.0.0.1] max_num_of_ostreams : 30 (../src/amf/amf-sm.c:745)

Please let me know if anyone is able to connect  UE in TDD mode.

**Reply by smunaut** [reply]
Yeah, I was able to connect but only when using 20 MHz cell width. For some reason 10 MHz wouldn't be recognized by my phones ...

**Reply by asheeshtripathi** [reply]
> Thanks for your feedback. Could you set log level to info and send us logs please?

Sure.

**Reply by asheeshtripathi** [reply]
> Yeah, I was able to connect but only when using 20 MHz cell width. For some reason 10 MHz wouldn't be recognized by my phones ...

The network is being detected by Telit modem and OnePlus8T both. I verified the RF transmission  with spectrum analyzer it is transmitting perfectly in 10 MHz config. But No attach.

**Reply by asheeshtripathi** [reply]
Hi @smunaut, I was able to proceed with the UE attach: registration and ip allocation to UE is successful. But UE connection is lost after GTP session is created. I am suspecting issue with the tunnel. I have srsRAN gNB and open5gs core co-located in the same machine. can you share the open5GS config and gnB config (for AMF and GTPU addresses)

**Reply by pgawlowicz** [reply]
You can try with the Open5gs docker image that we now provide in the srsRAN_Project. You can start as show here:
https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html#open5gs-core

Note that the tutorial is on connecting srsUE with the gNB, but Open5GS core and routing configurations can be reused. Please remember to populate the subscriber database using Open5gs Web GUI.

**Reply by asheeshtripathi** [reply]
Hi @pgawlowicz , Thanks for the reply. The issue was resolved long back, it was related to the  UE. One of the UEs that I used initially would first attach, and once PDU session created, it detached; after some time (60-120 seconds), another PDU session was created, and UE worked fine. With other UEs, I did not face any issues. I have tested srsRAN with both the dockerized version and non-dockerized Open5GS core.

**Reply by pgawlowicz** [reply]
Great! Thank you for the feedback.
